WordPress/Classes/Utility/Languages.swift

```diff
@@ -68,26 +68,29 @@ class Languages : NSObject
     /// If the language is not supported, it returns 1 (English).
     ///
     func deviceLanguageId() -> NSNumber {
-        if let exactMatch = languageWithSlug(deviceLanguageCode) {
-            return exactMatch.languageId
-        }
-        let languageTag = deviceLanguageCode.componentsSeparatedByString("-")[0]
-        if languageTag != deviceLanguageCode,
-            let partialMatch = languageWithSlug(languageTag) {
-            return partialMatch.languageId
+        let variants = LanguageTagVariants(string: deviceLanguageCode)
+        for variant in variants {
+            if let match = self.languageWithSlug(variant) {
+                return match.languageId
+            }
```

Comment:
If you're talking about `find`, it's a slightly different case: we don't want the first variant that matches the predicate, but we want to transform it and return the first non-nil.

A more functional approach would be:

```swift
return variants.lazy.flatMap({ self.languageWithSlug($0) }).first ?? 1
```

I think I tried something like that, but I don't remember if I hit some compiler errors or just found the current version more readable.